src/UI/Context.hs

```diff
@@ -1,17 +1,29 @@
+{-# LANGUAGE DeriveTraversable #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
 module UI.Context
 ( Context
+, Pixels(..)
 , runContext
 ) where
 
 import           Control.Carrier.Reader
 import           Control.Effect.Lift
 import qualified Control.Exception.Lift as E
 import           Control.Monad.IO.Class.Lift
+import           Data.Functor.Identity
+import           Foreign.Storable
+import           GL.Type as GL
+import           GL.Uniform
 import           Graphics.GL.Core41
 import           SDL
 
 type Context = GLContext
 
+newtype Pixels a = Pixels { getPixels :: a }
```

Comment:
We now have useful names for the two different kinds of pixels: `Context.Pixels` and `Window.Pixels`. The former represent the pixels as the context/buffer sees them, the latter as the UI/user sees them.